BJS2-84363-sms_notification

Linempy opened the pull request:
Изменения в userService для реализации SMS нотификаций

Linempy committed:
BJS2-84363-sms_notification
- "перевел" логи в RecommendationRequestServiceImpl
- поправил немного стиль кода
- подключил Redis
- сделал отправку ивента в нужный топик по Redis
- добавил поле в UserViewDto

RinatBeybutov commented on src/main/java/school/faang/user_service/entity/user/User.java:
```diff
@@ -36,11 +37,11 @@
 @NoArgsConstructor
 @AllArgsConstructor
 @Entity
+@Data
```
дата на энтити - это мега плохая вещь. Добавь файл чекстайла проверку на одновременное наличие аннотаций Entity и Data в одном классе. Если есть совпадение, то выбрасывать ошибку при проверке стиля

RinatBeybutov commented on src/main/java/school/faang/user_service/publisher/RecommendationRequestedEventPublisher.java:
```diff
@@ -0,0 +1,39 @@
+@RequiredArgsConstructor
+public class RecommendationRequestedEventPublisher {
+
+    private final RedisTemplate<String, Object> redisTemplate;
```
можно написать вот так:
RedisTemplate<String, RecommendationRequestedEvent>

RinatBeybutov commented on src/main/java/school/faang/user_service/publisher/RecommendationRequestedEventPublisher.java:
```diff
@@ -0,0 +1,39 @@
+    public void publish(RecommendationRequestedEvent event) {
+        String json = null;
+        try {
+            json = objectMapper.writeValueAsString(event);
```
тогда вот здесь не понадобится конвертировать в строку

RinatBeybutov commented on src/main/java/school/faang/user_service/publisher/RecommendationRequestedEventPublisher.java:
```diff
@@ -0,0 +1,39 @@
+            throw new RuntimeException(e);
+        }
+        log.info("Ивент отправлен в топик {}", recommendationRequestTopic);
+        redisTemplate.convertAndSend(recommendationRequestTopic, json);
```
проверь, что возвращает метод convertAndSend. Мб там future и нужно поправить, чтобы лог позитивный писался только после окончания future

RinatBeybutov commented on src/main/java/school/faang/user_service/service/recommendation/RecommendationRequestServiceImpl.java:
```diff
@@ -50,7 +53,8 @@ public RecommendationRequestViewDto create(RecommendationRequestCreateDto create
 
-        log.info("Recommendation request created successfully. ID: {}", request.getId());
+        log.info("Запрос рекомендации успешно создан. ID: {}", request.getId());
+        publisher.publish(new RecommendationRequestedEvent(requesterId, receiver.getId(), request.getId()));
```
писать в брокер в одной транзакции с сохранением в базу - не самая удачная идея. У тебя может быть проблема сетевая с редисом и их-ха этого отменится транзакция с базой. Подумай, как это можно решить

RinatBeybutov commented on src/main/java/school/faang/user_service/service/recommendation/RecommendationRequestServiceImpl.java:
```diff
@@ -65,56 +69,57 @@ public List<RecommendationRequestViewDto> getByFilters(RecommendationRequestFilt
                 RequestStatus.ACCEPTED,
-                "This user can't accepted request",
-                "Only pending requests can be accepted"
+                "Данный пользователь не может принять запрос на рекомендацию",
```
лучше вынести в константу класса

Linempy committed:
BJS2-84363-sms_notification
- поправил замечания
- изменил десериализацию на Jackson2Json в конфигурации Redis
- добавил интерфейс для publisher: EventPublisher

Linempy committed:
BJS2-84363-sms_notification
- добавил в checkstyle 2 новых правила
- поправил тесты

Linempy committed:
BJS2-84363-sms_notification
- checkstyle наше все

RinatBeybutov commented on config/checkstyle/checkstyle.xml:
```diff
@@ -33,7 +33,22 @@
+        <!-- Проверка SQL в нижнем регистре -->
+        <module name="RegexpSinglelineJava">
+            <property name="id" value="sqlLowerCaseCheck"/>
+            <property name="format" value="@Query.*select.*from"/>
```
нужно немного расширить список слов:
select|insert|update|delete|from|where|join|on|and|or|group by|order by|having

RinatBeybutov commented:
сделай эту правку, разреши конфликт и мержить будем

Linempy committed:
Merge branch 'manticore-master-stream11' into manticore11/BJS2-84363-sms_notification

Linempy committed:
BJS2-84316-recommend_analytics
- дополнил проверку на регистр sql-запросов в checkstyle
